Mathlib/CategoryTheory/MorphismProperty/Composition.lean

```diff
@@ -61,19 +61,25 @@ instance Pi.containsIdentities {J : Type w} {C : J → Type u}
     (pi W).ContainsIdentities :=
   ⟨fun _ _ => MorphismProperty.id_mem _ _⟩
 
-/-- A morphism property is `StableUnderComposition` if the composition of two such morphisms
-still falls in the class. -/
-def StableUnderComposition (P : MorphismProperty C) : Prop :=
-  ∀ ⦃X Y Z⦄ (f : X ⟶ Y) (g : Y ⟶ Z), P f → P g → P (f ≫ g)
-#align category_theory.morphism_property.stable_under_composition CategoryTheory.MorphismProperty.StableUnderComposition
+/-- A morphism property satisfies `IsStableUnderComposition` if the composition of
+two such morphisms still falls in the class. -/
+class IsStableUnderComposition (P : MorphismProperty C) : Prop :=
+  comp_mem {X Y Z} (f : X ⟶ Y) (g : Y ⟶ Z) : P f → P g → P (f ≫ g)
+#align category_theory.morphism_property.stable_under_composition CategoryTheory.MorphismProperty.IsStableUnderComposition
+
+lemma comp_mem (W : MorphismProperty C) [W.IsStableUnderComposition]
+    {X Y Z : C} (f : X ⟶ Y) (g : Y ⟶ Z) (hf : W f) (hg : W g) : W (f ≫ g) :=
+  IsStableUnderComposition.comp_mem f g hf hg
 
-theorem StableUnderComposition.op {P : MorphismProperty C} (h : StableUnderComposition P) :
-    StableUnderComposition P.op := fun _ _ _ f g hf hg => h g.unop f.unop hg hf
-#align category_theory.morphism_property.stable_under_composition.op CategoryTheory.MorphismProperty.StableUnderComposition.op
+instance IsStableUnderComposition.op {P : MorphismProperty C} [P.IsStableUnderComposition] :
+    P.op.IsStableUnderComposition where
+  comp_mem f g hf hg := P.comp_mem g.unop f.unop hg hf
+#align category_theory.morphism_property.stable_under_composition.op CategoryTheory.MorphismProperty.IsStableUnderComposition.op
 
-theorem StableUnderComposition.unop {P : MorphismProperty Cᵒᵖ} (h : StableUnderComposition P) :
-    StableUnderComposition P.unop := fun _ _ _ f g hf hg => h g.op f.op hg hf
-#align category_theory.morphism_property.stable_under_composition.unop CategoryTheory.MorphismProperty.StableUnderComposition.unop
+theorem IsStableUnderComposition.unop {P : MorphismProperty Cᵒᵖ} [P.IsStableUnderComposition] :
```

Comment:
This can be an instance as well I think.

Reply:
Thanks!